openedx/core/lib/xblock_services/video_config.py

```diff
@@ -0,0 +1,209 @@
+"""
+Video Configuration Service for XBlock runtime.
+
+This service provides video-related configuration and feature flags
+that are specific to the edx-platform implementation
+for the extracted video block in xblocks-contrib repository.
+"""
+
+import logging
+from typing import Optional
+
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.keys import UsageKeyV2
+
+from openedx.core.djangoapps.video_config.models import (
+    CourseYoutubeBlockedFlag,
+    HLSPlaybackEnabledFlag,
+)
+from openedx.core.djangoapps.video_config.toggles import TRANSCRIPT_FEEDBACK
+from openedx.core.djangoapps.video_pipeline.config.waffle import DEPRECATE_YOUTUBE
+from xmodule.exceptions import NotFoundError
+from openedx.core.lib.xblock_services.video_config_utils import (
+    get_public_video_url,
+    is_public_sharing_enabled,
+)
+from organizations.api import get_course_organization
+
+
+log = logging.getLogger(__name__)
+
+
+class VideoConfigService:
+    """
+    Service for providing video-related configuration and feature flags.
+
+    This service abstracts away edx-platform specific functionality
+    that the Video XBlock needs, allowing the Video XBlock to be
+    extracted to a separate repository.
+    """
+
+    def __init__(self, course_id: Optional[CourseKey] = None):
```

Comment:
Great to see new type annotations, thank you 👍🏻  Some feedback:

* For `Optional[Foo]` types, you can also annotate them using `Foo | None`, for example `CourseKey | None`. It's the same effect, but it reads a little nicer and you don't need to import `Optional`.
* Please ensure that every method parameter and return in this module are annotated
* Please add this module to `files =` in `mypy.ini` so that it actually gets checked in CI.